src/chain_sync/tipset_syncer.rs

```diff
@@ -695,9 +693,18 @@ where
 
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         loop {
-            match self.as_mut().tipset_tasks.as_mut().poll_next(cx) {
-                Poll::Ready(Some(Ok(_))) => continue,
-                Poll::Ready(Some(Err(e))) => return Poll::Ready(Err(e)),
+            match self.as_mut().tipset_tasks.poll_join_next(cx) {
+                Poll::Ready(Some(Ok(Ok(_)))) => continue,
+                Poll::Ready(Some(Ok(Err(e)))) => return Poll::Ready(Err(e)),
+                Poll::Ready(Some(Err(e))) => {
+                    _ = {
+                        if e.is_panic() {
+                            std::panic::resume_unwind(e.into_panic());
```

Comment:
Let's use [try_into_panic](https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html#method.try_into_panic). I'm fine with not printing the task id in case of cancellation.

Reply:
Fixed